// es-abstract/es2021.js
'use strict';

/* eslint global-require: 0 */
// https://262.ecma-international.org/12.0/#sec-abstract-operations
var ES2021 = {
	'Abstract Equality Comparison': require('es-abstract/2021/AbstractEqualityComparison'),
	'Abstract Relational Comparison': require('es-abstract/2021/AbstractRelationalComparison'),
	'Strict Equality Comparison': require('es-abstract/2021/StrictEqualityComparison'),
	abs: require('es-abstract/2021/abs'),
	AddEntriesFromIterable: require('es-abstract/2021/AddEntriesFromIterable'),
	AddToKeptObjects: require('es-abstract/2021/AddToKeptObjects'),
	AdvanceStringIndex: require('es-abstract/2021/AdvanceStringIndex'),
	ApplyStringOrNumericBinaryOperator: require('es-abstract/2021/ApplyStringOrNumericBinaryOperator'),
	ArrayCreate: require('es-abstract/2021/ArrayCreate'),
	ArraySetLength: require('es-abstract/2021/ArraySetLength'),
	ArraySpeciesCreate: require('es-abstract/2021/ArraySpeciesCreate'),
	AsyncFromSyncIteratorContinuation: require('es-abstract/2021/AsyncFromSyncIteratorContinuation'),
	AsyncIteratorClose: require('es-abstract/2021/AsyncIteratorClose'),
	BigInt: require('es-abstract/2021/BigInt'),
	BigIntBitwiseOp: require('es-abstract/2021/BigIntBitwiseOp'),
	BinaryAnd: require('es-abstract/2021/BinaryAnd'),
	BinaryOr: require('es-abstract/2021/BinaryOr'),
	BinaryXor: require('es-abstract/2021/BinaryXor'),
	ByteListBitwiseOp: require('es-abstract/2021/ByteListBitwiseOp'),
	ByteListEqual: require('es-abstract/2021/ByteListEqual'),
	Call: require('es-abstract/2021/Call'),
	Canonicalize: require('es-abstract/2021/Canonicalize'),
	CanonicalNumericIndexString: require('es-abstract/2021/CanonicalNumericIndexString'),
	CharacterRange: require('es-abstract/2021/CharacterRange'),
	clamp: require('es-abstract/2021/clamp'),
	ClearKeptObjects: require('es-abstract/2021/ClearKeptObjects'),
	CloneArrayBuffer: require('es-abstract/2021/CloneArrayBuffer'),
	CodePointAt: require('es-abstract/2021/CodePointAt'),
	CodePointsToString: require('es-abstract/2021/CodePointsToString'),
	CompletePropertyDescriptor: require('es-abstract/2021/CompletePropertyDescriptor'),
	CompletionRecord: require('es-abstract/2021/CompletionRecord'),
	CopyDataProperties: require('es-abstract/2021/CopyDataProperties'),
	CreateAsyncFromSyncIterator: require('es-abstract/2021/CreateAsyncFromSyncIterator'),
	CreateDataProperty: require('es-abstract/2021/CreateDataProperty'),
	CreateDataPropertyOrThrow: require('es-abstract/2021/CreateDataPropertyOrThrow'),
	CreateHTML: require('es-abstract/2021/CreateHTML'),
	CreateIterResultObject: require('es-abstract/2021/CreateIterResultObject'),
	CreateListFromArrayLike: require('es-abstract/2021/CreateListFromArrayLike'),
	CreateMethodProperty: require('es-abstract/2021/CreateMethodProperty'),
	CreateRegExpStringIterator: require('es-abstract/2021/CreateRegExpStringIterator'),
	DateFromTime: require('es-abstract/2021/DateFromTime'),
	DateString: require('es-abstract/2021/DateString'),
	Day: require('es-abstract/2021/Day'),
	DayFromYear: require('es-abstract/2021/DayFromYear'),
	DaysInYear: require('es-abstract/2021/DaysInYear'),
	DayWithinYear: require('es-abstract/2021/DayWithinYear'),
	DefinePropertyOrThrow: require('es-abstract/2021/DefinePropertyOrThrow'),
	DeletePropertyOrThrow: require('es-abstract/2021/DeletePropertyOrThrow'),
	DetachArrayBuffer: require('es-abstract/2021/DetachArrayBuffer'),
	EnumerableOwnPropertyNames: require('es-abstract/2021/EnumerableOwnPropertyNames'),
	FlattenIntoArray: require('es-abstract/2021/FlattenIntoArray'),
	floor: require('es-abstract/2021/floor'),
	FromPropertyDescriptor: require('es-abstract/2021/FromPropertyDescriptor'),
	Get: require('es-abstract/2021/Get'),
	GetGlobalObject: require('es-abstract/2021/GetGlobalObject'),
	GetIterator: require('es-abstract/2021/GetIterator'),
	GetMethod: require('es-abstract/2021/GetMethod'),
	GetOwnPropertyKeys: require('es-abstract/2021/GetOwnPropertyKeys'),
	GetPromiseResolve: require('es-abstract/2021/GetPromiseResolve'),
	GetPrototypeFromConstructor: require('es-abstract/2021/GetPrototypeFromConstructor'),
	GetSubstitution: require('es-abstract/2021/GetSubstitution'),
	GetV: require('es-abstract/2021/GetV'),
	GetValueFromBuffer: require('es-abstract/2021/GetValueFromBuffer'),
	HasOwnProperty: require('es-abstract/2021/HasOwnProperty'),
	HasProperty: require('es-abstract/2021/HasProperty'),
	HourFromTime: require('es-abstract/2021/HourFromTime'),
	InLeapYear: require('es-abstract/2021/InLeapYear'),
	InstanceofOperator: require('es-abstract/2021/InstanceofOperator'),
	IntegerIndexedElementGet: require('es-abstract/2021/IntegerIndexedElementGet'),
	IntegerIndexedElementSet: require('es-abstract/2021/IntegerIndexedElementSet'),
	InternalizeJSONProperty: require('es-abstract/2021/InternalizeJSONProperty'),
	Invoke: require('es-abstract/2021/Invoke'),
	IsAccessorDescriptor: require('es-abstract/2021/IsAccessorDescriptor'),
	IsArray: require('es-abstract/2021/IsArray'),
	IsBigIntElementType: require('es-abstract/2021/IsBigIntElementType'),
	IsCallable: require('es-abstract/2021/IsCallable'),
	IsCompatiblePropertyDescriptor: require('es-abstract/2021/IsCompatiblePropertyDescriptor'),
	IsConcatSpreadable: require('es-abstract/2021/IsConcatSpreadable'),
	IsConstructor: require('es-abstract/2021/IsConstructor'),
	IsDataDescriptor: require('es-abstract/2021/IsDataDescriptor'),
	IsDetachedBuffer: require('es-abstract/2021/IsDetachedBuffer'),
	IsExtensible: require('es-abstract/2021/IsExtensible'),
	IsGenericDescriptor: require('es-abstract/2021/IsGenericDescriptor'),
	IsIntegralNumber: require('es-abstract/2021/IsIntegralNumber'),
	IsNoTearConfiguration: require('es-abstract/2021/IsNoTearConfiguration'),
	IsPromise: require('es-abstract/2021/IsPromise'),
	IsPropertyKey: require('es-abstract/2021/IsPropertyKey'),
	IsRegExp: require('es-abstract/2021/IsRegExp'),
	IsSharedArrayBuffer: require('es-abstract/2021/IsSharedArrayBuffer'),
	IsStringPrefix: require('es-abstract/2021/IsStringPrefix'),
	IsUnclampedIntegerElementType: require('es-abstract/2021/IsUnclampedIntegerElementType'),
	IsUnsignedElementType: require('es-abstract/2021/IsUnsignedElementType'),
	IsValidIntegerIndex: require('es-abstract/2021/IsValidIntegerIndex'),
	IsWordChar: require('es-abstract/2021/IsWordChar'),
	IterableToList: require('es-abstract/2021/IterableToList'),
	IteratorClose: require('es-abstract/2021/IteratorClose'),
	IteratorComplete: require('es-abstract/2021/IteratorComplete'),
	IteratorNext: require('es-abstract/2021/IteratorNext'),
	IteratorStep: require('es-abstract/2021/IteratorStep'),
	IteratorValue: require('es-abstract/2021/IteratorValue'),
	LengthOfArrayLike: require('es-abstract/2021/LengthOfArrayLike'),
	MakeDate: require('es-abstract/2021/MakeDate'),
	MakeDay: require('es-abstract/2021/MakeDay'),
	MakeTime: require('es-abstract/2021/MakeTime'),
	max: require('es-abstract/2021/max'),
	min: require('es-abstract/2021/min'),
	MinFromTime: require('es-abstract/2021/MinFromTime'),
	modulo: require('es-abstract/2021/modulo'),
	MonthFromTime: require('es-abstract/2021/MonthFromTime'),
	msFromTime: require('es-abstract/2021/msFromTime'),
	NewPromiseCapability: require('es-abstract/2021/NewPromiseCapability'),
	NormalCompletion: require('es-abstract/2021/NormalCompletion'),
	Number: require('es-abstract/2021/Number'),
	NumberBitwiseOp: require('es-abstract/2021/NumberBitwiseOp'),
	NumberToBigInt: require('es-abstract/2021/NumberToBigInt'),
	NumericToRawBytes: require('es-abstract/2021/NumericToRawBytes'),
	ObjectDefineProperties: require('es-abstract/2021/ObjectDefineProperties'),
	OrdinaryCreateFromConstructor: require('es-abstract/2021/OrdinaryCreateFromConstructor'),
	OrdinaryDefineOwnProperty: require('es-abstract/2021/OrdinaryDefineOwnProperty'),
	OrdinaryGetOwnProperty: require('es-abstract/2021/OrdinaryGetOwnProperty'),
	OrdinaryGetPrototypeOf: require('es-abstract/2021/OrdinaryGetPrototypeOf'),
	OrdinaryHasInstance: require('es-abstract/2021/OrdinaryHasInstance'),
	OrdinaryHasProperty: require('es-abstract/2021/OrdinaryHasProperty'),
	OrdinaryObjectCreate: require('es-abstract/2021/OrdinaryObjectCreate'),
	OrdinarySetPrototypeOf: require('es-abstract/2021/OrdinarySetPrototypeOf'),
	OrdinaryToPrimitive: require('es-abstract/2021/OrdinaryToPrimitive'),
	PromiseResolve: require('es-abstract/2021/PromiseResolve'),
	QuoteJSONString: require('es-abstract/2021/QuoteJSONString'),
	RawBytesToNumeric: require('es-abstract/2021/RawBytesToNumeric'),
	RegExpCreate: require('es-abstract/2021/RegExpCreate'),
	RegExpExec: require('es-abstract/2021/RegExpExec'),
	RequireObjectCoercible: require('es-abstract/2021/RequireObjectCoercible'),
	SameValue: require('es-abstract/2021/SameValue'),
	SameValueNonNumeric: require('es-abstract/2021/SameValueNonNumeric'),
	SameValueZero: require('es-abstract/2021/SameValueZero'),
	SecFromTime: require('es-abstract/2021/SecFromTime'),
	Set: require('es-abstract/2021/Set'),
	SetFunctionLength: require('es-abstract/2021/SetFunctionLength'),
	SetFunctionName: require('es-abstract/2021/SetFunctionName'),
	SetIntegrityLevel: require('es-abstract/2021/SetIntegrityLevel'),
	SetTypedArrayFromArrayLike: require('es-abstract/2021/SetTypedArrayFromArrayLike'),
	SetTypedArrayFromTypedArray: require('es-abstract/2021/SetTypedArrayFromTypedArray'),
	SetValueInBuffer: require('es-abstract/2021/SetValueInBuffer'),
	SpeciesConstructor: require('es-abstract/2021/SpeciesConstructor'),
	SplitMatch: require('es-abstract/2021/SplitMatch'),
	StringCreate: require('es-abstract/2021/StringCreate'),
	StringGetOwnProperty: require('es-abstract/2021/StringGetOwnProperty'),
	StringIndexOf: require('es-abstract/2021/StringIndexOf'),
	StringPad: require('es-abstract/2021/StringPad'),
	StringToBigInt: require('es-abstract/2021/StringToBigInt'),
	StringToCodePoints: require('es-abstract/2021/StringToCodePoints'),
	substring: require('es-abstract/2021/substring'),
	SymbolDescriptiveString: require('es-abstract/2021/SymbolDescriptiveString'),
	TestIntegrityLevel: require('es-abstract/2021/TestIntegrityLevel'),
	thisBigIntValue: require('es-abstract/2021/thisBigIntValue'),
	thisBooleanValue: require('es-abstract/2021/thisBooleanValue'),
	thisNumberValue: require('es-abstract/2021/thisNumberValue'),
	thisStringValue: require('es-abstract/2021/thisStringValue'),
	thisSymbolValue: require('es-abstract/2021/thisSymbolValue'),
	thisTimeValue: require('es-abstract/2021/thisTimeValue'),
	ThrowCompletion: require('es-abstract/2021/ThrowCompletion'),
	TimeClip: require('es-abstract/2021/TimeClip'),
	TimeFromYear: require('es-abstract/2021/TimeFromYear'),
	TimeString: require('es-abstract/2021/TimeString'),
	TimeWithinDay: require('es-abstract/2021/TimeWithinDay'),
	TimeZoneString: require('es-abstract/2021/TimeZoneString'),
	ToBigInt: require('es-abstract/2021/ToBigInt'),
	ToBigInt64: require('es-abstract/2021/ToBigInt64'),
	ToBigUint64: require('es-abstract/2021/ToBigUint64'),
	ToBoolean: require('es-abstract/2021/ToBoolean'),
	ToDateString: require('es-abstract/2021/ToDateString'),
	ToIndex: require('es-abstract/2021/ToIndex'),
	ToInt16: require('es-abstract/2021/ToInt16'),
	ToInt32: require('es-abstract/2021/ToInt32'),
	ToInt8: require('es-abstract/2021/ToInt8'),
	ToIntegerOrInfinity: require('es-abstract/2021/ToIntegerOrInfinity'),
	ToLength: require('es-abstract/2021/ToLength'),
	ToNumber: require('es-abstract/2021/ToNumber'),
	ToNumeric: require('es-abstract/2021/ToNumeric'),
	ToObject: require('es-abstract/2021/ToObject'),
	ToPrimitive: require('es-abstract/2021/ToPrimitive'),
	ToPropertyDescriptor: require('es-abstract/2021/ToPropertyDescriptor'),
	ToPropertyKey: require('es-abstract/2021/ToPropertyKey'),
	ToString: require('es-abstract/2021/ToString'),
	ToUint16: require('es-abstract/2021/ToUint16'),
	ToUint32: require('es-abstract/2021/ToUint32'),
	ToUint8: require('es-abstract/2021/ToUint8'),
	ToUint8Clamp: require('es-abstract/2021/ToUint8Clamp'),
	TrimString: require('es-abstract/2021/TrimString'),
	Type: require('es-abstract/2021/Type'),
	TypedArrayCreate: require('es-abstract/2021/TypedArrayCreate'),
	TypedArraySpeciesCreate: require('es-abstract/2021/TypedArraySpeciesCreate'),
	UnicodeEscape: require('es-abstract/2021/UnicodeEscape'),
	UTF16EncodeCodePoint: require('es-abstract/2021/UTF16EncodeCodePoint'),
	UTF16SurrogatePairToCodePoint: require('es-abstract/2021/UTF16SurrogatePairToCodePoint'),
	ValidateAndApplyPropertyDescriptor: require('es-abstract/2021/ValidateAndApplyPropertyDescriptor'),
	ValidateAtomicAccess: require('es-abstract/2021/ValidateAtomicAccess'),
	ValidateIntegerTypedArray: require('es-abstract/2021/ValidateIntegerTypedArray'),
	ValidateTypedArray: require('es-abstract/2021/ValidateTypedArray'),
	WeakRefDeref: require('es-abstract/2021/WeakRefDeref'),
	WeekDay: require('es-abstract/2021/WeekDay'),
	WordCharacters: require('es-abstract/2021/WordCharacters'),
	YearFromTime: require('es-abstract/2021/YearFromTime')
};

module.exports = ES2021;
